diff --git a/src/python/epicscorelibs/config.py b/src/python/epicscorelibs/config.py
index 2c307453a..70b3d8f21 100644
--- a/src/python/epicscorelibs/config.py
+++ b/src/python/epicscorelibs/config.py
@@ -106,7 +106,7 @@ def _makebuild():
 
     if OS_CLASS=='Darwin':
         build['CPPFLAGS'] += [('darwin', None)]
-        build['CXXFLAGS'] += ['-std=c++98', '-stdlib=libstdc++']
+        build['CXXFLAGS'] += ['-std=c++11', '-stdlib=libc++']
 
     if OS_CLASS!='WIN32':
         build['CPPFLAGS'] += [('UNIX', None)]
